catalog/lib/widgets/app_text_area.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:catalog/catalog.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:catalog/widgets/app_text_fields.dart';
+
+class AppTextArea extends StatefulWidget {
+  final TextEditingController textController;
+  final String? labelText;
+  final String? hintText;
+  final void Function(String text) onChange;
+  final bool isOptional;
+  final int maxLength;
+  final int minLines;
+  final int maxLines;
+
+  const AppTextArea({
+    required this.textController,
+    required this.labelText,
+    required this.hintText,
+    required this.onChange,
+    required this.maxLength,
+    required this.minLines,
+    required this.maxLines,
+    this.isOptional = false,
+    super.key,
+  });
+
+  @override
+  State<AppTextArea> createState() => _AppTextAreaState();
+}
+
+class _AppTextAreaState extends State<AppTextArea> {
+  int _characterCount = 0;
+
+  @override
+  Widget build(BuildContext context) => Column(
+        children: [
+          AppTextField(
```

Comment:
If we give a textfield decoration input the maxlines field and it "transforms" to a text Area.
Like this: 
```
TextField(
  keyboardType: TextInputType.multiline,
  maxLines: 20,
)
```
The same goes with the maximum number of characters it has a [property](https://api.flutter.dev/flutter/material/TextField/maxLength.html) for the number of characters